client/my-sites/site-settings/seo-settings/form.jsx

```diff
@@ -332,7 +336,7 @@ export class SeoForm extends React.Component {
 				<QuerySiteSettings siteId={ siteId } />
 				{ siteId && <QueryJetpackPlugins siteIds={ [ siteId ] } /> }
 				{ siteIsJetpack && <QueryJetpackModules siteId={ siteId } /> }
-				{ ( isSitePrivate || isSiteHidden ) && hasSiteSeoFeature( selectedSite ) && (
+				{ ( isSitePrivate || isSiteHidden ) && showAdvancedSeo && (
```

Comment:
`showAdvancedSeo` includes a check for `hasSiteSeoFeature( selectedSite )`, while making sure we don't show this warning on `jetpack_free` sites. We could leave this as-is though, if there are other types of sites that should show it.